tests/p2p/integration_test_helpers.py

```diff
@@ -56,10 +56,16 @@ class FakeAsyncChainDB(AsyncChainDB):
         return self.persist_trie_data_dict(*args, **kwargs)
 
 
+async def coro_import_block(chain, block, perform_validation=True):
+    # Be nice and yield control to give other coroutines a chance to run before us as
+    # importing a block is a very expensive operation.
+    await asyncio.sleep(0)
+    return chain.import_block(block, perform_validation=perform_validation)
```

Comment:
Am I correct that from the networking process's perspective this will be run in an executor (via the proxy, via the `async_method` util) and then within the db process, this will block during execution?

Reply:
Well, that actually happens because of https://github.com/ethereum/py-evm/pull/638/files#diff-763cbc43185be6a4e5239e86e9b71e1dR141. Here we're looking at the `FakeAsyncChain` implementations, whose async methods are actually blocking and hence the `await sleep(0)` at the beginning

Reply:
Oh, lol, didn't even notice what file I was in.  👍